### ISSUE TITLE
iOS crash when trying to sync

### ISSUE THREAD

**Issue by isthisish** [etesync/ios#45]
Hello,

I just downloaded EteSync today. I have a bunch of calendars (Gmail, Nextcloud, iOS) on my iOS, so it was hard to figure out which one I wanted to set to sync. I chose one, turned off iCloud, and the app crashed. It’s happened now for the last 2 hours. The log is below:

```
Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:21:1111
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:1229:154
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:1256:4493
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1740
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
n@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:898
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1273
c@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:205
b@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:1623
_@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:488
u@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1257
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1309
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:3681
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:1256:3511
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:1254:1711
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1740
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
n@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:898
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1047
f@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:155
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:882
y@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:661
C@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:1025
callImmediates@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:3100
callImmediates@[native code]
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:3247
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:1283
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:2939
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:1253
value@[native code]
value@[native code]
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Fetching collection Contacts
[W] XXX i9S8Yme3dVDZgYUt-KIwFk1fVJ-UvM5X 0 0
[I] Starting sync (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Fetching collection Calendar
[W] XXX CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84 0 0
[I] Starting sync (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[D] local:  (68851806-CF40-4EBB-B1CD-87BCD58695D8), cardDAV: Address Book (D2192177-F39C-4654-AF2C-65BC0F6A9350), cardDAV: Address Book (7C559DC6-23AE-4752-8831-8A5813833850), cardDAV: Address Book (85E20AEF-B509-415E-94BC-F38315DA2083), cardDAV: Contacts (EB93A9D5-5C6C-4211-A7F2-0E787C5DAACD), cardDAV: Recently contacted (21CC044A-6B89-4575-869D-F438F0837F75)
[D] caldav: IH (311A1DA8-F09D-49DD-84B8-236F3724E0BF), caldav: H&M (FDBB969A-78CB-4FF3-8DA3-E2099BA6EC53), caldav: Sunburnt (88EA79F9-5041-4FBF-BDA8-4788D44987ED), caldav: Nextcloud (66E20004-5F7A-4065-8E30-F78FD9C24B72), local: Default (F0AF15D4-3F6D-49E6-A666-617748D27A53), caldav: Kid (E2CC983D-C178-4F78-ACF6-81AB8784B1A0)
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[D] local:  (68851806-CF40-4EBB-B1CD-87BCD58695D8), cardDAV: Address Book (D2192177-F39C-4654-AF2C-65BC0F6A9350), cardDAV: Address Book (7C559DC6-23AE-4752-8831-8A5813833850), cardDAV: Address Book (85E20AEF-B509-415E-94BC-F38315DA2083), cardDAV: Contacts (EB93A9D5-5C6C-4211-A7F2-0E787C5DAACD), cardDAV: Recently contacted (21CC044A-6B89-4575-869D-F438F0837F75)
[D] caldav: $$ (311A1DA8-F09D-49DD-84B8-236F3724E0BF), caldav: $$$ (FDBB969A-78CB-4FF3-8DA3-E2099BA6EC53), caldav: $$$$$$$ (88EA79F9-5041-4FBF-BDA8-4788D44987ED), caldav: $$$$$$$$$ (66E20004-5F7A-4065-8E30-F78FD9C24B72), local: Default (F0AF15D4-3F6D-49E6-A666-617748D27A53), caldav: $$$ (E2CC983D-C178-4F78-ACF6-81AB8784B1A0)
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Fetching collection Contacts
[W] XXX i9S8Yme3dVDZgYUt-KIwFk1fVJ-UvM5X 40 40
[W] XXX i9S8Yme3dVDZgYUt-KIwFk1fVJ-UvM5X 7 7
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Deleting SUwgsStoZpwBjdPu9WgihzX2LUIbW7aW
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier E4A867F1-7410-42A4-949F-6911EB512458 not found
[I] ConnectionfInfo: true wifi
```

**Comment by tasn** [comment]
Could you try selecting the collection to sync after turning off icloud? (e.g. go to the app's settings, disable calendar sync, and then choose a calendar account again).

**Comment by isthisish** [comment]
Hi Tom,

I did, but no luck. Still same error.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by isthisish** [comment]
I’ve gone ahead and deleted Calendar5 by Readdle, which was bringing all the other accounts together. Still, no luck

```
Calendar: failed to find selected source F0AF15D4-3F6D-49E6-A666-617748D27A53. Please contact developers.
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:1256:1772
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1740
p@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:423
n@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:898
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:112:1047
f@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:155
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:41:882
y@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:661
C@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:1025
callImmediates@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:114:3100
callImmediates@[native code]
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:3247
https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:1283
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:2939
value@https://expo.etesync.com/release/5/bundles/ios-cec52f7a219fd2e7a28390e01ddba26b.js:27:1253
value@[native code]
value@[native code]
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pulling changes (Calendars)
[I] Pulling CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Applying changes. Current stoken: tJnVH_AVwZ5QxaAafG12kWCWPtzb0i6n, last one: null
[I] Finished sync (Calendars)
[I] Starting sync (Tasks)
[I] Creating H--gTd-xxNBXdtMemxaYT5qLtHQGfD6H
[C] Error: Calendar 'My Tasks' could not be saved: Error Domain=EKErrorDomain Code=24 "That account does not support reminders." UserInfo={NSLocalizedDescription=That account does not support reminders.}
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[I] ConnectionfInfo: true wifi
[I] ConnectionfInfo: true wifi
[I] ConnectionfInfo: false none
[I] Disconnected, aborting sync
[I] Disconnected, aborting sync
[I] ConnectionfInfo: true wifi
[I] ConnectionfInfo: true wifi
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pushing changes (Calendars)
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier 2DCCEEEF-FDDE-431A-A1DF-636B77F2795D not found
[I] ConnectionfInfo: true wifi
[I] Permissions status for calendar: granted
[I] ConnectionfInfo: true wifi
[I] Permissions status for reminders: granted
[I] Permissions status for contacts: granted
[I] Permissions status for userFacingNotifications: granted
[D] local:  (68851806-CF40-4EBB-B1CD-87BCD58695D8), cardDAV: Address Book (D2192177-F39C-4654-AF2C-65BC0F6A9350), cardDAV: Address Book (7C559DC6-23AE-4752-8831-8A5813833850), cardDAV: Address Book (85E20AEF-B509-415E-94BC-F38315DA2083), cardDAV: Contacts (EB93A9D5-5C6C-4211-A7F2-0E787C5DAACD), cardDAV: Recently contacted (21CC044A-6B89-4575-869D-F438F0837F75)
[D] caldav: IH (311A1DA8-F09D-49DD-84B8-236F3724E0BF), caldav: H&M (FDBB969A-78CB-4FF3-8DA3-E2099BA6EC53), caldav: Sunburnt (88EA79F9-5041-4FBF-BDA8-4788D44987ED), caldav: Nextcloud (66E20004-5F7A-4065-8E30-F78FD9C24B72), local: Default (F0AF15D4-3F6D-49E6-A666-617748D27A53), caldav: Mailbox (2BEC2ED6-8DF7-411C-99D9-D9361EAD1CEB)
[I] Fetching collection list
[I] Starting sync (Calendars)
[I] Syncing journal list (Calendars)
[I] Updating CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[I] Pushing CEvBFfryDJn0DBJkRCc6RBWX_vXFeZ84
[C] Error: Calendar with identifier 2DCCEEEF-FDDE-431A-A1DF-636B77F2795D not found
[I] ConnectionfInfo: true wifi
```

**Comment by Petersoj** [comment]
I am having a similar issue. When I import a large `.ics` file on the EteSync Web UI (which succeeds) and then try to sync it to a 'localhost' CalDAV account, the app crashes every time I try to sync. It appears that the crash is happening whenever the EteSync app is trying to write a lot of new events to the local iOS calendar app after syncing all the newly imported events from the EteSync server. The app does successfully write some events to the local iOS calendar app as I see some imported events, but crashes mid-sync (which causes the EteSync app to think that there were new local changes so it now pushes those events to the server which causes duplicate events). This happens both on my iPhone and my iPad (although the iPad EteSync app was able to write way more events to the local iOS calendar app than the iPhone EteSync app (maybe a memory issue?)).

**Comment by articlecat** [comment]
Same issue here. It looks like a memory issue, since EteSync does manage to sync a couple hundred of my contacts in spite of the crash, although the remainder never makes it onto the device. When I wiped the contacts, uninstalled and reinstalled EteSync, and tried syncing again, EteSync managed to sync the same number of contacts before crashing.

**Comment by scott-8** [comment]
I'm also having the same issue, Etesync starts syncing and crashes partway through. It successfully writes a portion of the contacts, but the journal in the app is incomplete. Syncing grabs new changes but does nothing about the missing contacts.

**Comment by UnixPhonez** [comment]
i'm getting the same error, i was hoping to use this app to sync my spouses and my calendar together..

**Comment by demided** [comment]
@isthisish did you get get any resolution for this?

**Comment by solarkraft** [comment]
I get a very similar issue, every time it's syncing a specific calendar collection. 
Only on the mobile app, on web and the CalDAV bridge it works fine. 
I can't delete the calendar, because it has important stuff in it. 
But now I can't use Etesync on my phone at all because this error crashes the sync. 
This is bad! I depend on it!

**Comment by solarkraft** [comment]
@isthisish Why did you close the issue? Did you find a solution?

**Comment by solarkraft** [comment]
This error consistently appears the SECOND time the refresh button is pressed. 

Newly created events (on an other device) appear in the iOS app but not the local calendar and events created in the local calendar don't appear in the app. 

It remains an issue even after logging out, renaming or deleting the affected calendar locally and logging back in. 

Sync on other platforms works just fine (no warnings in the console on the web app).

**Comment by solarkraft** [comment]
For context why I'm trying so hard to avoid deleting everything: This issue has been existing for a long while and I fear losing events that were created on the device since then. 

Code containing the string "Calendar: failed to find selected source" can be found at: https://github.com/etesync/ios/blob/966c112bd6a92daaee2ab128cea29118a13188d8/src/sync/legacy/SyncManagerCalendar.ts#L38

Deleting the local calendar and messing with the sync settings led me to the error "Calendar 'my_calendar' could not be saved: Error Domain=EKErrorDomain Code=17 "That account does not allow calendars to be added or removed.". 

The line `if (storeState.settings.ranWizrd) {` led me to reinstall the app (the calendars are technically local to the device, so they won't be lost) to re-run the setup wizard. 
This initially yielded a new error: "Calendar with identifier <UUID> not found". 
However after restarting the app, everything seems to be working as it should!. 

During the process I did re-create a calendar, which may or may not have helped. To back it up I used the sketchy looking and 1$ costing app [Export calendar](https://apps.apple.com/us/app/export-calendars/id1510047412), so now Maciej Hofstede probably knows what I'm doing next weekend.
